### PR TITLE
[MAINTENANCE] Move `mostly` to correct Expectation classes 

### PR DESCRIPTION
**Pull request by NathanFarmer**
- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: great_expectations/expectations/expectation.py [hunk 1]
```diff
@@ -1506,7 +1506,6 @@ class BatchExpectation(Expectation, ABC):
     batch_id: Union[str, None] = None
     row_condition: Union[str, None] = None
     condition_parser: Union[str, None] = None
-    mostly: Mostly = 1.0
```
Comment (NathanFarmer) [inline great_expectations/expectations/expectation.py:1509]: This was wrong. `mostly` does not apply to non-map Expectations, but those Expectations inherit from `BatchExpectation`.

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 2]
```diff
@@ -753,7 +753,6 @@ class TestColumnIdentifiers:
     @pytest.mark.parametrize(
         "expectation_type",
         [
-            "expect_column_to_exist",
```
Comment (josectobar) [inline tests/datasource/fluent/integration/test_sql_datasources.py]: Why are we removing this expectation type?

Comment (NathanFarmer) [reply tests/datasource/fluent/integration/test_sql_datasources.py]: `mostly` is not valid for this Expectation

**Review by josectobar — CHANGES_REQUESTED** [review]
Is there a way we can fix the test without having to reduce the number of expectation types we're testing?

**Review by josectobar — APPROVED** [review]
Looks great thanks!

**Pull request merged**